kubespawner/reflector.py

```diff
@@ -29,12 +28,46 @@ class PodReflector(SingletonConfigurable):
         """
     )
 
-    pods = Dict(
+    resources = Dict(
         {},
         help="""
-        Dictionary of pod names to Pod objects.
+        Dictionary of resource names to the appropriate resource objects.
+
+        This can be accessed across threads safely.
+        """
+    )
+
+    kind = Unicode(
+        'resource',
+        help="""
+        Human readable name for kind of object we're watching for.
+
+        Used for diagnostic messages.
+        """
+    )
+
+    list_method_name = Unicode(
+        None,
+        allow_none=True,
+        help="""
+        Name of function (on apigroup respresented by `api_group_name`) that is to be called to list resources.
 
-        This can be directly accessed from multiple threads.
+        This will be passed a namespace & a label selector. You most likely want something
+        of the form list_namespaced_<resource> - for example, `list_namespaced_pod` will
+        give you a PodReflector.
+
+        This must be set by a subclass.
+        """
+    )
+
+    api_group_name = Unicode(
+        'CoreV1Api',
+        allow_none=False,
```

Comment:
No need to specify `allow_none=False`. In general, I don't think we should use `allow_none` on Unicode traits unless there's a specific, odd case for empty strings meaning something specific.

Reply:
Yeah, I think this is one of the cases where code I've written (default to None for default) has been different from code you've written (default to "", not allow None). Since it looks like your style is more prevelant in the Jupyter community & I can't make a compelling case for my style, I'll switch to yours from now!